src/crimson/common/operation.h

```diff
@@ -502,15 +502,22 @@ class PipelineHandle {
   }
 
   template <typename OpT, typename T>
-  seastar::future<>
-  do_enter(T &stage, typename T::BlockingEvent::template Trigger<OpT>&& t) {
-    auto fut = t.maybe_record_blocking(stage.enter(t), stage);
-    exit();
-    return std::move(fut).then(
-      [this, t=std::move(t)](auto &&barrier_ref) mutable {
+  std::optional<seastar::future<>>
+  do_enter_maybe_sync(T &stage, typename T::BlockingEvent::template Trigger<OpT>&& t) {
```

Comment:
Can you please add a comment or a commit message here explanation here?

Reply:
nit: do_enter_maybe_block?

Reply:
I can't quite follow what the intent here is.  This function needs a comment explaining what do_enter_maybe_sync does and what the return value means.